services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/application/Application.kt

```diff
@@ -7,29 +7,35 @@ enum class ApplicationType {
     CABLE_REPORT,
 }
 
+/** Interface to enable Application and CableReportWithoutHanke handling equivalently. */
+interface HaitatonApplication {
```

Comment:
The alternative would be to have two separate validators that share the code validating the application data. That one would be conceptually simpler, but would of course involve more code.

I think the two approaches are six of one, half a dozen of the other. Except maybe for the naming. `HaitatonApplication` sounds like it's a special case of `Application` when it's the other way round. It's hard to find a good name for the interface, since it should be `Application`, but that one is already taken and for a good reason. The best name I can come up with is something like `BaseApplication`. The other approach would of course circumvent this naming problem.

Reply:
I guess baseapplication is ok. 

I ended up using an interface after initially having a separate validator with shared validation functions. Mainly due to tests, the implementation had something like 700 lines of code changes, most of it duplication. 

If there is not much of a difference which approach is better, I'd prefer this one.

Reply:
Renamed interface to BaseApplication.